app/controllers/incomeSources/add/AddBusinessAddressController.scala

```diff
@@ -86,9 +87,18 @@ class AddBusinessAddressController @Inject()(authenticate: AuthenticationPredica
     }
   }
 
-  def handleSubmitRequest(isAgent: Boolean, id: Option[String])(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+  def getRedirectUrl(isAgent: Boolean, isChange: Boolean): String = {
+    if (isChange) {
```

Comment:
* worth considering using pattern matching as nested if/else usually hard to read;

Reply:
ok i'll do this